docs/key-issues/cross_platform.md

```diff
@@ -0,0 +1,33 @@
+# Cross-platform installation
+
+The historical assumption of compilation is that the platform where the code is
+compiled will be the same as the platform where the final code will be executed
+(if not literally the same machine, then at least one that is CPU and ABI
+compatible at the operating system level). This is a reasonable assumption for
+most desktop projects; However, for mobile platforms, this isn't the case.
+
+On mobile platforms, an app is compiled on a desktop platform, and transferred
+to the mobile device (or a simulator) for testing. The compiler is not executed
+on device. Therefore, it must be possible to build a binary artefact for a CPU
+architecture and a ABI that is different the platform that is running the
+compiler.
+
+A microcosm of this problem exists on macOS as a result of the Apple Silicon
+transition. Most CI systems don't provide native ARM hardware, but most
+developers will still want ARM64-compatible build artefacts. Apple has provided
+the tools compile [fat binaries](multiple_architectures.md) on x86_64 hardware;
+however, in this case, the host platform (macOS on x86_64) will still be one of
+the outputs of the compilation process. For mobile platforms, the computer that
+compiles the code will not be able to execute the code that has been compiled.
```

Comment:
I guess we're all viewing the world through our own little prism, but leading with mobile and using "microcosm" for osx-arm is kinda counter to my experience in Python packaging (even though that might be a very justified POV in beeware).

I think if we add a cross-compilation page (and certain bits and pieces are already sprinkled here and there, e.g. in `background/compilation_concepts.md`), we should aim to have a wider view on the topic.

I agree though that it probably merits its own page. It's a very tricky problem at scale (because so many packages have build instructions that just assume everything runs on native hardware), as we see constantly in conda-forge for osx-arm, linux-aarch, linux-ppc64le, etc.

Reply:
My framing of this as a "microcosm" is on the basis that the problem is broadly similar, but less complicated. macOS only needs to deal with 2 platforms; one of those platforms will always be the host platform, so the build tooling will run on host; and the compiler can always work in a single pass. For iOS, I've had to deal with up to 5 platforms simultaneously for iOS, and *none* of them were native, so none of the build tools run natively, and it almost always requires multiple compiler passes.